src/mesh/mod.rs

```diff
@@ -114,23 +110,53 @@ impl Mesh {
                                 &vertices_collection[cast_usize(v3)],
                             );
 
-                            normals_collection.push(face_normal);
-
+                            normals_collection_sharp.push(face_normal);
                         }
-                        // FIXME: once we add other kinds of faces, they must panic here
-                        // _ => panic!("Face must be a triangle, we just created it"),
                     }
                 }
+                assert_eq!(normals_collection_sharp.len(), faces_collection_sharp.len());
+                assert_eq!(
+                    normals_collection_sharp.capacity(),
+                    faces_collection_sharp.len()
+                );
+                (faces_collection_sharp, normals_collection_sharp)
+            }
+            NormalStrategy::Smooth => {
+                let faces_collection_smooth: Vec<_> = faces
+                    .into_iter()
+                    .map(|(i1, i2, i3)| TriangleFace::new_separate(i1, i2, i3, i1, i2, i3))
```

Comment:
You don't need `TriangleFace::new_separate` here. `new` is enough, because the indices are shared.

Reply:
ok